packages/dataviews/src/view-table.js

```diff
@@ -70,15 +70,49 @@ function HeaderMenu( { dataView, header } ) {
 	}
 	const sortedDirection = header.column.getIsSorted();
 
-	let filter;
+	let filter, filterInView;
+	const otherFilters = [];
 	if ( header.column.columnDef.type === ENUMERATION_TYPE ) {
-		filter = {
-			field: header.column.columnDef.id,
-			elements: header.column.columnDef.elements || [],
-		};
+		let columnOperators = header.column.columnDef.filterBy?.operators;
+		if ( ! columnOperators || ! Array.isArray( columnOperators ) ) {
+			columnOperators = [ OPERATOR_IN, OPERATOR_NOT_IN ];
+		}
+		const operators = columnOperators.filter( ( operator ) =>
+			[ OPERATOR_IN, OPERATOR_NOT_IN ].includes( operator )
+		);
+		if ( operators.length >= 0 ) {
```

Comment:
Just noticing that this is the same pattern of validation as seen in `./filters.js`

* * *

Also, is this condition right? It will always evaluate to true, as it's the length of an array.

```js
 operators.length >= 0
```

Reply:
This is a bug, thanks for catching! Fix at https://github.com/WordPress/gutenberg/pull/57048

Reply:
On the topic of code reuse: there's an opportunity to reuse some code between the filters.js and the view-table.js. I gave it a try at https://github.com/WordPress/gutenberg/pull/56514 though it wasn't merged. I'm happy to pair on this and take a look at how it could be improved.

Reply:
Don't worry, I think premature consolidation is a danger. I was just pointing out the pattern for later. :)

Reply:
Prompted by this conversation, I've taken a look and consolidated little details so all the filter implementations are mostly lookalike (but still maintaining the little UI/interaction differences for each one) https://github.com/WordPress/gutenberg/pull/57059

It's not as ambitious as 56514 was, but I've found some bugs with the column filters by doing this exercise, so it's worth having anyway.